ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java

```diff
@@ -452,4 +459,43 @@ public EventDispatcher getEventDispatcher() {
   public void sendAccessibilityEvent(int tag, int eventType) {
     mUIImplementation.sendAccessibilityEvent(tag, eventType);
   }
+
+  private DisplayMetrics getDisplayMetrics() {
+    Context context = getReactApplicationContext();
+
+    DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
+    WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
+    Display display = wm.getDefaultDisplay();
+
+    // Get the real display metrics if we are using API level 17 or higher.
+    // The real metrics include system decor elements (e.g. soft menu bar).
+    //
+    // See: http://developer.android.com/reference/android/view/Display.html#getRealMetrics(android.util.DisplayMetrics)
+    if (Build.VERSION.SDK_INT >= 17){
+      try {
+        display.getRealMetrics(displayMetrics);
+      } catch (Exception e) {
+        throw new RuntimeException("Error getting real dimensions for API level >= 17", e);
+      }
+
+    } else {
+      // For 14 <= API level <= 16, we need to invoke getRawHeight and getRawWidth to get the real dimensions.
+      // Since react-native only supports API level 16+ we don't have to worry about other cases.
+      //
+      // Reflection exceptions are rethrown at runtime.
+      //
+      // See: http://stackoverflow.com/questions/14341041/how-to-get-real-screen-height-and-width/23861333#23861333
+      try {
+        Method mGetRawH = Display.class.getMethod("getRawHeight");
+        Method mGetRawW = Display.class.getMethod("getRawWidth");
+        displayMetrics.widthPixels = (Integer) mGetRawW.invoke(display);
+        displayMetrics.heightPixels = (Integer) mGetRawH.invoke(display);
+      } catch (Exception e) {
```

Comment:
Can you catch the specific Exception types? You should be able to `or` the class names together and keep it as just one catch clause